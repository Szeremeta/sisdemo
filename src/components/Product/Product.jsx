import React from 'react';
import PropTypes from 'prop-types';

const Product = ({
    id,
    description,
    expiryDate,
    category,
    price,
    special,
    handleDelete,
    handleEdit,
}) => {
    return (
        <div className="product">
            <h2>{description}</h2>
            <h4>{category}</h4>
            <h1>&pound;{price}</h1>
            {expiryDate && (<p>Expiry Date: {expiryDate}</p>)}
            {special && (<div className="product-special">ON SPECIAL!!</div>)}
            <button onClick={()=>handleDelete(id)}>Delete</button>
            <button onClick={()=>handleEdit(id)}>Edit</button>
        </div>
    );
}

Product.propTypes = {
    id: PropTypes.number.isRequired,
    description: PropTypes.string.isRequired,
    expiryDate: PropTypes.string,
    category: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
    special: PropTypes.bool,
    handleDelete: PropTypes.func,
    handleEdit: PropTypes.func,
}

Product.defaultProps = {
    expiryDate: '',
    special: false,
    handleDelete: () => { },
    handleEdit: () => { },
}

export default Product;