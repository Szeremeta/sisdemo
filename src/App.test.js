import React from 'react';
import { render, screen } from '@testing-library/react';
import Product from './components/Product/Product';

describe('Product Component', () => {
  const props = {
    id: 1,
    description: 'Test Product',
    expiryDate: '2024-04-30',
    category: 'Test Category',
    price: 10.99,
    special: true,
  };

  it('renders product information correctly', () => {
    render(<Product {...props} />);
    expect(screen.getByText('Test Product'));
    expect(screen.getByText('Test Category'));
    expect(screen.getByText('Â£10.99'));
    expect(screen.getByText('ON SPECIAL!!'));
    expect(screen.getByText('Expiry Date: 2024-04-30'));
  });

  it('matches snapshot', () => {
    const wrapper = render(<Product {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});