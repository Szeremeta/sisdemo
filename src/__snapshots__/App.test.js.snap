// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Component matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="product"
      >
        <h2>
          Test Product
        </h2>
        <h4>
          Test Category
        </h4>
        <h1>
          £
          10.99
        </h1>
        <p>
          Expiry Date: 
          2024-04-30
        </p>
        <div
          class="product-special"
        >
          ON SPECIAL!!
        </div>
        <button>
          Delete
        </button>
        <button>
          Edit
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="product"
    >
      <h2>
        Test Product
      </h2>
      <h4>
        Test Category
      </h4>
      <h1>
        £
        10.99
      </h1>
      <p>
        Expiry Date: 
        2024-04-30
      </p>
      <div
        class="product-special"
      >
        ON SPECIAL!!
      </div>
      <button>
        Delete
      </button>
      <button>
        Edit
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
